DOC: Clarify single-segment arrays in np reference (#16011)
numpy/numpy: doc/source/reference/arrays.ndarray.rst

single-segment, memory layouts, in which every part of the
memory block can be accessed by some combination of the indices.

.. note::

>     `Contiguous arrays` and `single-segment arrays` are synonymous. They have been
>     used interchangeably in the numpy docs and have the same meaning.

Reviewer: ```suggestion
    `Contiguous arrays` and `single-segment arrays` are synonymous
    and are used interchangeably throughout the documentation.
```
